compose.yaml

```
@@ -0,0 +1,344 @@
+# Local development-only compose file. This is not suitable for production!
+
+x-janus-common: &janus_common
+  image: us-west2-docker.pkg.dev/divviup-artifacts-public/janus/janus_aggregator:0.7.7
+  restart: always
+  healthcheck:
+    test: ["CMD", "/bin/sh", "-c", "wget http://0.0.0.0:8000/healthz -O - >/dev/null"]
+  depends_on:
+    janus_1_migrate:
+      condition: service_completed_successfully
+    janus_2_migrate:
+      condition: service_completed_successfully
+
+x-janus-migrate: &janus_migrate
+  image: us-west2-docker.pkg.dev/divviup-artifacts-public/janus/janus_db_migrator:0.7.7
+  command:
+    - migrate
+    - run
+    - --source
+    - /migrations
+    - --connect-timeout
+    - "60"
+  depends_on:
+    postgres:
+      condition: service_started
+
+x-janus-environment: &janus_environment
+  RUST_LOG: info
+  PGPASSWORD: postgres
+  DATASTORE_KEYS: 1B6szboUUtMfyrLsIVE20g
+  AGGREGATOR_API_AUTH_TOKENS: "0000"
+
+services:
+  postgres:
+    image: docker.io/library/postgres:16
+    ports:
+      - "5432:5432"
+    restart: always
+    environment:
+      POSTGRES_PASSWORD: postgres
+    configs:
+      - source: postgres_init
+        target: /docker-entrypoint-initdb.d/postgres_init.sql
+
+  divviup_api_migrate:
+    build:
+      context: .
+      args:
+        RUST_PROFILE: dev
+        # This isn't strictly required for migrations, but it allows reusing one
+        # container image for both this and the divviup_api service.
+        RUST_FEATURES: integration-testing
+    entrypoint:
+      - /migration
+      - up
+    environment:
+      DATABASE_URL: postgres://postgres:postgres@postgres:5432/divviup
+    depends_on:
+      postgres:
+        condition: service_started
+    develop:
+      watch:
+        - path: migration
+          action: rebuild
+          ignore:
+            - README.md
+
+  divviup_api:
+    build:
+      context: .
+      args:
+        RUST_PROFILE: dev
+        RUST_FEATURES: integration-testing
+    ports:
+      - "8080:8080"
+    environment:
+      RUST_LOG: info
+      AUTH_URL: https://auth.example
+      AUTH_CLIENT_ID: fake
+      AUTH_CLIENT_SECRET: fake
+      SESSION_SECRETS: AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
+      DATABASE_ENCRYPTION_KEYS: heHIWC_qxHzdglI0VgLopQ
+      AUTH_AUDIENCE: https://app.example
+      APP_URL: http://localhost:8081
+      DATABASE_URL: postgres://postgres:postgres@postgres:5432/divviup
+      API_URL: http://localhost:8080
+      POSTMARK_TOKEN: fake
+      EMAIL_ADDRESS: the@from.address
+    depends_on:
+      divviup_api_migrate:
+        condition: service_completed_successfully
+    develop:
+      watch:
+        - path: src/
+          action: rebuild
+
+  divviup_api_vite:
+    build:
+      context: .
+      dockerfile: Dockerfile.vite
```

If we don't need to do anything else with this image, we could skip the extra Dockerfile
```suggestion
    image: node:alpine
```

---

Unfortunately, `develop` doesn't work when you have a `image` instead of a `build` directive. I'll leave a comment.